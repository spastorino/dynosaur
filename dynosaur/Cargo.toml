[package]
name = "dynosaur"
version = "0.2.0"
authors = [
    "Santiago Pastorino <spastorino@gmail.com>",
    "Tyler Mandry <tmandry@google.com>",
    "Niko Matsakis <rust@nikomatsakis.com>",
]
categories = ["asynchronous", "no-std"]
keywords = ["async", "trait", "impl"]
description = "Dynamic dispatch for return position impl traits and async in Rust"
documentation = "https://docs.rs/dynosaur"
license = "MIT OR Apache-2.0"
repository = "https://github.com/spastorino/dynosaur"
edition = "2021"
rust-version = "1.75"

[dependencies]
dynosaur_derive = { version = "0.2", path = "../dynosaur_derive" }

[dev-dependencies]
tokio = { workspace = true }
trait-variant = { workspace = true }
ui_test = "0.28"

[[example]]
name = "next"

[[example]]
name = "trait_variant"

[[test]]
name = "ui"
harness = false
