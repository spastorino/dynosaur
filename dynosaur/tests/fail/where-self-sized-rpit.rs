use dynosaur::dynosaur;

#[dynosaur(DynMyTrait)]
trait MyTrait {
    fn foo(&mut self) -> impl ::core::future::Future<Output = ()>
    where
        Self: Sized;
    async fn bar(&mut self);
}

fn main() {}
