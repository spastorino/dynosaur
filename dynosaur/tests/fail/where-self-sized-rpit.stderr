error: where Self: Sized is unsupported
 --> tests/fail/where-self-sized-rpit.rs:5:5
  |
5 | /     fn foo(&mut self) -> impl ::core::future::Future<Output = ()>
6 | |     where
7 | |         Self: Sized;
  | |____________________^

error: aborting due to 1 previous error

